### PR TITLE
fix(config): stop accidentally requiring region for ES

### PR DESCRIPTION
**Pull request by lukesteensen**
This is an at-least-temporary fix to #1160

### CODE REVIEW

File: src/sinks/elasticsearch.rs [hunk 1]
```diff
@@ -128,9 +131,10 @@ impl ElasticSearchCommon {
             format!("Basic {}", base64::encode(token.as_bytes()))
         });
 
-        let region: Option<Region> = match config.region {
-            Some(ref region) => Some(region.try_into()?),
-            None => None,
+        let region: Option<Region> = match config.region.clone().try_into() {
```
Comment (bruceg) [inline src/sinks/elasticsearch.rs]: It's a little ugly, but this can be `(&config.region).try_into()` to avoid the clone.

**Comment by bruceg** [comment]
Note there is another bug with the handling of `host` as well -- when `provider = "aws"`, the hostname is still required, even though it shouldn't be (ref all the other AWS sinks).

**Comment by lukesteensen** [comment]
Good point. Would you mind opening an issue for that? I'd like to get this merged ASAP but we can circle back around and maybe come up with a better solution that covers both of those cases.

**Comment by bruceg** [comment]
Definitely, go ahead.

**Pull request merged**